### PR TITLE
Highlevelv1

### PR DESCRIPTION
**Pull request by AFrancani**

### CODE REVIEW

**Comment by AFrancani** [comment]
Suggestion of refactorings
- [ ] delete the `helpers.py`
- [ ] rename the exceptions to remove `Doc`
- [ ] rename `test_client.py` to `test_api.py`
- [ ] rename `highlevel.py` (and test file) to `client.py`

**Comment by ak0ska** [comment]
* [ ] Add changelog
* [ ] Add test for main client (just that it can be instantiated, there's no typo)
* [ ] Linting

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+## Unreleased
+
+### Added
+
+ * A storage service client `hbp_service_client.storage_service.client` with a higher level abstraction, adding the following methods:
+   * TODO when the docstrings are added.
+
+### Changed
+ * The module `hbp_service_client.document_service` was renames to `hbp_service_client.storage_service`
```
Comment (AFrancani) [inline CHANGELOG.md]: renames -> renamed

File: hbp_service_client/storage_service/client.py [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+        Raises:
+            StorageArgumentException: Invalid arguments
+            StorageForbiddenException: Server response code 403
+            StorageNotFoundException: Server response code 404
+            StorageException: other 400-600 error codes
+        '''
+
+        self.__validate_storage_path(path)
+        entity = self.api_client.get_entity_by_query(path=path)
+        assert entity['entity_type'] == 'file'
```
Comment (AFrancani) [inline hbp_service_client/storage_service/client.py]: I think we should use the `StorageArgumentException` here, what do you think @ak0ska ?

File: hbp_service_client/storage_service/client.py [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+            StorageForbiddenException: Server response code 403
+            StorageNotFoundException: Server response code 404
+            StorageException: other 400-600 error codes
+        '''
+        self.__validate_storage_path(path, projects_allowed=False)
+        parent_metadata = self.get_parent(path)
+        self.api_client.create_folder(path.split('/')[-1], parent_metadata['uuid'])
+        #no return necessary, function succeeds or we would have thrown an exception before this point.
+
+    def upload_file(self, local_file, dest_path, mimetype, md5check=False):
```
Comment (AFrancani) [inline hbp_service_client/storage_service/client.py]: We can remove the md5check parameter as it's not used

File: requirements_extension_tests.txt [hunk 4]
```diff
@@ -1,6 +1,7 @@
 PyHamcrest==1.9.0
-mock==1.0.1
-requests_mock==1.0.0
+mock==2.0.0
 mock-matchers==0.0.1
-sniffer==0.4.0
 HTTPretty==0.8.14
+green==2.11.0
```
Comment (AFrancani) [inline requirements_extension_tests.txt]: As discussed, we can probably remove `green`

File: requirements_extension_tests.txt [hunk 5]
```diff
@@ -1,6 +1,7 @@
...
+pytest
```
Comment (AFrancani) [inline requirements_extension_tests.txt]: Shouldn't we pin versions?

File: hbp_service_client/storage_service/client.py (hunk 2 above)
Comment (ak0ska) [reply hbp_service_client/storage_service/client.py]: Done

File: hbp_service_client/storage_service/client.py (hunk 3 above)
Comment (ak0ska) [reply hbp_service_client/storage_service/client.py]: Well spotted. Removed.

File: requirements_extension_tests.txt (hunk 4 above)
Comment (ak0ska) [reply requirements_extension_tests.txt]: Removed.

File: requirements_extension_tests.txt (hunk 5 above)
Comment (ak0ska) [reply requirements_extension_tests.txt]: Versions unpinned

**Pull request merged**